fiftyone/factory/repos/delegated_operation.py

```diff
@@ -312,7 +312,7 @@ def update_run_state(
 
         doc = self._collection.find_one_and_update(
             filter={"_id": _id},
-            update=[update],
+            update=update,
```

Comment:
I believe this should be an array (aggregation pipeline). Otherwise passing a dictionary will be interpreted as the new doc so you'd essentially be setting the doc to have a new field "$set" with value equal to the dictionary value

Reply:
You may be thinking of `find_one_and_replace()`. This change works and tests pass.
It's more standard to pass a dict with the updates. It's possible, though uncommon, to use an agg pipeline, but I didn't see why that was necessary.

Reply:
Alternatively if you don't specify `$set` and pass a [replacement document](https://www.mongodb.com/docs/manual/reference/method/db.collection.update/#std-label-update-method-replacement-document) instead, then it would behave as you say.

Reply:
https://www.mongodb.com/docs/manual/reference/method/db.collection.findOneAndUpdate/

I'm just going off of the mongo docs

Reply:
But yeah the alternative is to remove set

Reply:
And yes the test `test_updates_progress` tests for this. After talking with Ibrahim I am also making an update and adding a test for a specific corner case that's broken.